File: performance_talk/array.c
#include <math.h>
#include <stdio.h>
#include <sys/time.h>

int main() {
  printf("Start\n");
  struct timeval start,stop;
  float start_time, stop_time;
  gettimeofday(&start, NULL);
  int size = 100000;
  float array[size];
  float ans[size];
  printf("Here\n");
  for (int i=0; i<size; i++) {
    ans[i] = tan(array[i]);
  }
  gettimeofday(&stop, NULL);
  start_time = (start.tv_sec * 1000000) + (start.tv_usec);
  stop_time = (stop.tv_sec * 1000000) + (stop.tv_usec);
  printf("time: %f\n", (stop_time - start_time));
  printf("End\n");
}

File: performance_talk/array.py
import time
import math

t0 = time.time()

size = 100000000
A = list(range(size))
B = list(range(size))
for i in range(size):
    B[i] = math.tan(A[i])

t1 = time.time()
print(t1 - t0)
